lib/fastly_nsq.rb

```diff
@@ -1,62 +1,56 @@
+# frozen_string_literal: true
+
 require 'nsq'
-require 'fastly_nsq/consumer'
-require 'fastly_nsq/fake_backend'
-require 'fastly_nsq/listener'
-require 'fastly_nsq/message'
-require 'fastly_nsq/messenger'
-require 'fastly_nsq/producer'
-require 'fastly_nsq/strategy'
-require 'fastly_nsq/tls_options'
-require 'fastly_nsq/version'
+require 'concurrent'
+require 'fc'
+require 'set'
+require 'logger'
+require 'forwardable'
 
 module FastlyNsq
-  module_function
+  ConfigurationError = Class.new(StandardError)
+  NotConnectedError = Class.new(StandardError)
 
-  @preprocessor = nil
+  class << self
+    attr_accessor :channel, :preprocessor
+    attr_writer :logger
 
-  def channel=(channel)
-    @channel ||= channel
-  end
-
-  def logger=(logger)
-    strategy.logger = logger
-  end
+    def listen(topic, processor)
```

Comment:
should this not also take a `priority` at the very least?

Reply:
oooo good call.

Reply:
https://github.com/fastly/fastly_nsq/pull/65/commits/aa3c1e814679373e62a5d2d42a81b76633fb5177